packages/docs/docs/docs/migrate_from_1.x_to_2.x.md

```diff
@@ -0,0 +1,117 @@
+
+What is an EVM package? It is basically a piece of reusable code that has been deployed to the blockchain in the form of EVM bytecode. Anyone can create an EVM package, and your zOS projects will be able to connect to a vast ecosystem of pre-compiled, pre-deployed reusable code in the form of EVM packages.
+
+Similarly, as we realized what an EVM package was, we realized what it's not. In 1.x, your zOS project's contracts were always managed by two contracts: App and Package. This made your contracts upgradeable and gave them the ability to link to EVM packages (see [Contract architecture](https://docs.zeppelinos.org/docs/architecture.html) for more info). We realized that these two contracts made sense when it came to EVM packages, but didn't when it came to parts of your project that weren't meant to be reusable code, i.e. code that just needed to be upgradeable but not reusable. And so, in 2.x, upgradeability, and the ability to link to EVM packages were completely separated.
+
+In 2.x by default, your zOS project will not use any of its App or Package on-chain architecture unless you explicitly state that your code is meant to be reusable as an EVM package. When you init a zOS project, there is no architecture other than your project's code. ZeppelinOS simply uses the CLI to manage proxies for you. Now, if you intend your project to be a reusable EVM package, you can `publish` your code, and the CLI will seamlessly deploy the necessary App and Package architecture which will allow your code to exist as a discoverable deployed EVM package.
+
+## New commands
+
+2.x introduces a few new commands to its CLI which are described below:
+
+* `zos publish`
+* `zos check`
+* `zos set-admin`
+* `zos unlink`
+
+#### Publish
+
+```
+Usage: publish --network <network> [options]
+
+publishes your project to the selected network
+
+Options:
+  -n, --network <network>  network to be used
+  -f, --from <from>        specify transaction sender address
+  --timeout <timeout>      timeout in seconds for each blockchain transaction (defaults to 600s)
+  -h, --help               output usage information
+```
+
+As explained in the ["Terminology and fundamentals" section](https://docs.zeppelinos.org/docs/migrate_from_1.x_to_2.x.html#terminology-and-fundamentals), this command is what transforms your project to a reusable EVM package. It deploys the App and Package architecture and ensures that your code is discoverable as an EVM package.
```

Comment:
This paragraph and the next two should be on the guide that explains the push command.
It is WIP on my restructure task, so I can now just copy :)